span-normalizer/span-normalizer/src/test/java/org/hypertrace/core/spannormalizer/JaegerSpanNormalizerTest.java

```diff
@@ -160,6 +158,83 @@ public void testServiceNameAddedToEvent_excludeTenantIds() throws Exception {
     Assertions.assertNotNull(rawSpan2);
   }
 
+  @Test
+  public void testSpanDropCriterion() throws Exception {
+    String tenantId = "tenant-" + random.nextLong();
+    Map<String, Object> configs = new HashMap<>(getCommonConfig());
+    configs.putAll(
+        Map.of(
+            "processor",
+            Map.of("tenantIdTagKey", "tenant-key",
+                "spanDropCriterion", List.of("foo:bar,k1:v1"))));
+    JaegerSpanNormalizer normalizer = JaegerSpanNormalizer.get(ConfigFactory.parseMap(configs));
+    Process process = Process.newBuilder().setServiceName("testService").build();
+    Span span1 =
+        Span.newBuilder()
+            .setProcess(process)
+            .addTags(KeyValue.newBuilder().setKey("tenant-key").setVStr(tenantId).build())
+            .addTags(KeyValue.newBuilder().setKey("foo").setVStr("bar").build())
+            .build();
+    RawSpan rawSpan1 = normalizer.convert(span1);
+    Assertions.assertNotNull(rawSpan1);
+
+    Span span2 =
+        Span.newBuilder()
+            .setProcess(process)
+            .addTags(KeyValue.newBuilder().setKey("tenant-key").setVStr(tenantId).build())
+            .addTags(KeyValue.newBuilder().setKey("foo").setVStr("bar").build())
+            .addTags(KeyValue.newBuilder().setKey("k1").setVStr("v1").build())
+            .build();
+    RawSpan rawSpan2 = normalizer.convert(span2);
+    Assertions.assertNull(rawSpan2);
+  }
+
+  @Test
+  public void testDropSpanWithMultipleCriterion() throws Exception {
+    String tenantId = "tenant-" + random.nextLong();
+    Map<String, Object> configs = new HashMap<>(getCommonConfig());
+    configs.putAll(
+        Map.of(
+            "processor",
+            Map.of("tenantIdTagKey", "tenant-key",
+                "spanDropCriterion", List.of("foo:bar,k1:v1", "k2:v2"))));
+
+    JaegerSpanNormalizer normalizer = JaegerSpanNormalizer.get(ConfigFactory.parseMap(configs));
+    Process process = Process.newBuilder().setServiceName("testService").build();
+    Span span3 =
+        Span.newBuilder()
+            .setProcess(process)
+            .addTags(KeyValue.newBuilder().setKey("tenant-key").setVStr(tenantId).build())
+            .addTags(KeyValue.newBuilder().setKey("foo").setVStr("bar").build())
+            .addTags(KeyValue.newBuilder().setKey("k2").setVStr("v2").build())
+            .build();
+    RawSpan rawSpan3 = normalizer.convert(span3);
+    Assertions.assertNull(rawSpan3);
+  }
+
+  @Test
+  public void testDropSpanWithEmptyCriterion() throws Exception {
+    String tenantId = "tenant-" + random.nextLong();
+    Map<String, Object> configs = new HashMap<>(getCommonConfig());
+    configs.putAll(
+        Map.of(
+            "processor",
+            Map.of("tenantIdTagKey", "tenant-key",
```

Comment:
what is the tenantIdTagKey for ? are these criterion per tenant ?

Reply:
The criterion isn't per tenant. Thought about it but it made sense to keep them across the board.